libwallet/internal/vsp/vsp.go

```diff
@@ -53,127 +62,115 @@ func New(cfg Config) (*Client, error) {
 		return nil, err
 	}
 
+	if len(cfg.PubKey) == 0 {
+		return nil, errors.New("pubkey option not set")
+	}
+
 	if cfg.Wallet == nil {
-		return nil, fmt.Errorf("wallet option not set")
+		return nil, errors.New("wallet option not set")
+	}
+
+	if cfg.Params == nil {
+		return nil, errors.New("params option not set")
 	}
 
-	client := newClient(u.String(), cfg.PubKey, cfg.Wallet)
+	if cfg.Policy == nil {
+		return nil, errors.New("policy option not set")
+	}
+
+	client := &vspd.Client{
+		URL:    u.String(),
+		PubKey: cfg.PubKey,
+		Sign:   cfg.Wallet.SignMessage,
+		Log:    log,
+	}
 	client.Transport = &http.Transport{
 		DialContext: cfg.Dialer,
 	}
 
 	v := &Client{
-		Wallet: cfg.Wallet,
-		client: client,
+		wallet: cfg.Wallet,
+		policy: cfg.Policy,
+		Client: client,
 		jobs:   make(map[chainhash.Hash]*feePayment),
+		params: cfg.Params,
 	}
 	return v, nil
 }
 
+// SetAccountInfo set the account purchase information
+func (c *Client) SetAccountInfo(feeAcc, changeAcc int32) {
+	c.policy.FeeAcct = uint32(feeAcc)
+	c.policy.ChangeAcct = uint32(changeAcc)
+}
```

Comment:
Could separate processes try to set this info, leading to unexpected results? Why don't callers provide the policy to use, at the point the policy is needed?